packages/features/form-builder/Components.tsx

```diff
@@ -18,12 +17,25 @@ import {
   RadioField,
   EmailField,
   Tooltip,
+  InputField,
   Checkbox,
 } from "@calcom/ui";
 import { UserPlus, X } from "@calcom/ui/components/icon";
 
-import { ComponentForField } from "./FormBuilder";
-import type { fieldsSchema } from "./FormBuilderFieldsSchema";
+import { ComponentForField } from "./FormBuilderField";
+import { propsTypes } from "./propsTypes";
+import type { FieldType, variantsConfigSchema, fieldSchema } from "./schema";
+import { preprocessNameFieldDataWithVariant } from "./utils";
+
+export const isValidValueProp: Record<Component["propsType"], (val: unknown) => boolean> = {
+  boolean: (val) => typeof val === "boolean",
+  multiselect: (val) => val instanceof Array && val.every((v) => typeof v === "string"),
+  objectiveWithInput: (val) => (typeof val === "object" && val !== null ? "value" in val : false),
+  select: (val) => typeof val === "string",
+  text: (val) => typeof val === "string",
+  textList: (val) => val instanceof Array && val.every((v) => typeof v === "string"),
+  variants: (val) => (typeof val === "object" && val !== null) || typeof val === "string",
+};
```

Comment:
can't we use zod here ?

```
const booleanSchema = z.boolean();

const multiselectSchema = z.array(z.string());

const objectiveWithInputSchema = z.object({
  value: z.unknown()
});

const selectSchema = z.string();

const textSchema = z.string();

const textListSchema = z.array(z.string());

const variantsSchema = z.union([
  z.object({
    value: z.unknown()
  }),
  z.string()
]);
```

Reply:
That's a nice idea. Zod code here seems to be cleaner. Initially, these validations were complex and I wanted more flexibility. I didn't want to use superfine to implement custom errors. 

I am working on a PR over Form Builder to support making cancellationReason required, I will change it along with that